### PR TITLE
Flagging comments feature issue#411

### PR DESCRIPTION
**Pull request by Ayush0730**
## What is the change?
Added a feature to add flags to comments and if flags are more than or equal to three, then comment will be deleted automatically. Once this PR get merged I will be making a PR to restrict deletion of comments to writer(of comment) only.
This feature can be improved more such that one person will be able to flag a comment only one time but that requires a different approach.

## Related issue?
close: #411 

## How was it tested?
Locally

## Checklist:
Before you create this PR, confirm all the requirements listed below by checking the checkboxes `[x]`:

- [x] Have you followed the [Contribution Guidelines](https://github.com/ALPHAVIO/BlogSite/blob/master/CONTRIBUTING.md) while contributing.
- [x] Have you checked there aren't other open [Pull Requests](https://github.com/ALPHAVIO/BlogSite/pulls) for the same update/change?
- [] Did you lint your code before making the Pull Request?
- [] Have you made corresponding changes to the documentation?
- [x] Your submission doesn't break any existing feature.
- [x] Have you tested the code before submission?

## Screenshots or Video:

https://user-images.githubusercontent.com/68139755/115057551-d3ac4c00-9f01-11eb-82a2-c00c09500ca1.mp4

### CODE REVIEW

File: routes/post.router.js [hunk 1]
```diff
@@ -165,6 +166,41 @@ router.post('/posts/:postId/comments/:commentNum', auth, async (req, res) => {
     res.redirect(`/posts/${requestedPostId}`);
 });
 
+router.post(
+    '/posts/:postId/comments/:commentNum/flag',
+    auth,
+    async (req, res) => {
+        const isUser = !!req.user;
```
Comment (Pranav016) [inline routes/post.router.js:173]: `!!` used here ?

**Review by Pranav016 — CHANGES_REQUESTED** [review]
Maybe instead of a counter variable `flag=0`, you could use an array that stores the user name of the person flagging the comment. That way you can check from the array and prevent person from flagging again and use the length of the array as the number of times the comment got flagged. Let me know if this approach suites this usecase..

**Comment by Ayush0730** [comment]
> Maybe instead of a counter variable `flag=0`, you could use an array that stores the user name of the person flagging the comment. That way you can check from the array and prevent person from flagging again and use the length of the array as the number of times the comment got flagged. Let me know if this approach suites this use case..

File: routes/post.router.js (hunk 1 above)
Comment (Ayush0730) [reply routes/post.router.js:173]: It is for checking if user is Authenticated or not

**Comment by Ayush0730** [comment]
> Maybe instead of a counter variable `flag=0`, you could use an array that stores the user name of the person flagging the comment. That way you can check from the array and prevent person from flagging again and use the length of the array as the number of times the comment got flagged. Let me know if this approach suites this use case..

Done!!

**Review by Pranav016 — CHANGES_REQUESTED** [review]
changes look good, one thing. You checked the person has already flagged, now the flag button appears clicked or that individual ? I think for that you might have add this check in the post route as well other than the `/flag` route..

If you didn't get what I was trying to say then just add a demo video of the current changes and I'll explain..

**Comment by Ayush0730** [comment]
> changes look good, one thing. You checked the person has already flagged, now the flag button appears clicked or that individual ? I think for that you might have add this check in the post route as well other than the `/flag` route..
> 
> If you didn't get what I was trying to say then just add a demo video of the current changes and I'll explain..

For that, I would need to check for each comment in ejs file, post.ejs

**Comment by Ayush0730** [comment]
Please check if this is alright, there were many issues going on with my local branch. Please check for any mistakes, I may have made in committing and pushing

**Comment by Ayush0730** [comment]
Uploading bandicam 2021-04-17 23-16-32-669.mp4…

**Comment by Ayush0730** [comment]
https://user-images.githubusercontent.com/68139755/115121987-2efa3f00-9fd3-11eb-9a58-375f6c468ac7.mp4

Final feature demo video

**Comment by Pranav016** [comment]
>  bandicam.2021-04-17.23-16-32-669.mp4 
> Final feature demo video

Same person can't flag twice right ? So should we send some warning/error if the same person tries to flag again ?

**Comment by Ayush0730** [comment]
> > bandicam.2021-04-17.23-16-32-669.mp4
> > Final feature demo video
> 
> Same person can't flag twice right ? So should we send some warning/error if the same person tries to flag again ?

Well, flag button is disabled if the current user has already flagged that comment. So I don't think, this will ever happen

**Pull request merged**